sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -1987,6 +1987,18 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils with Te
     assert(errMsg2.contains("'lineSep' can contain only 1 character"))
   }
 
+  test("SPARK-26208: write and read empty data to csv file with header") {
+    withTempPath { path =>
+      val df1 = Seq.empty[(String, String)].toDF("x", "y")
```

Comment:
This will create a dataframe with one empty partition. I would check the case when there are more than 2 empty partitions, and each saved file has a header.

Reply:
You can do something like this:
```
val df = spark.range(10).repartition(2).filter(_ < 0)
```
```
scala> df.rdd.getNumPartitions
res0: Int = 2
```

Reply:
i can do that, but i think when i write it out and read it back in it will come back in as 1 partition (one part file with header) because of SPARK-23271. is that worth checking for?